src/models/relations-container.ts

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { Relations } from "./relations";
+import { EventType, RelationType } from "../@types/event";
+import { EventStatus, MatrixEvent, MatrixEventEvent } from "./event";
+import { EventTimelineSet } from "./event-timeline-set";
+import { MatrixClient } from "../client";
+import { Room } from "./room";
+
+export class RelationsContainer {
+    // A tree of objects to access a set of relations for an event, as in:
+    // this.relations[relatesToEventId][relationType][relationEventType]
+    private relations: {
+        [relatesToEventId: string]: {
+            [relationType: RelationType | string]: {
+                [eventType: EventType | string]: Relations;
+            };
+        };
+    } = {};
+
+    constructor(private readonly client: MatrixClient, private readonly room?: Room) {
+    }
+
+    /**
+     * Get a collection of relations to a given event in this timeline set.
+     *
+     * @param {String} eventId
+     * The ID of the event that you'd like to access relation events for.
+     * For example, with annotations, this would be the ID of the event being annotated.
+     * @param {String} relationType
+     * The type of relation involved, such as "m.annotation", "m.reference", "m.replace", etc.
+     * @param {String} eventType
+     * The relation event's type, such as "m.reaction", etc.
+     * @throws If <code>eventId</code>, <code>relationType</code> or <code>eventType</code>
+     * are not valid.
+     *
+     * @returns {?Relations}
+     * A container for relation events or undefined if there are no relation events for
+     * the relationType.
+     */
+    public getRelationsForEvent(
+        eventId: string,
+        relationType: RelationType | string,
+        eventType: EventType | string,
+    ): Relations | undefined {
+        return this.relations[eventId]?.[relationType]?.[eventType];
+    }
+
+    public getAllRelationsEventForEvent(eventId: string): MatrixEvent[] {
```

Comment:
```suggestion
    public getAllRelationsEventsForEvent(eventId: string): MatrixEvent[] {
```

.. or are they related events? :P